cli/azd/pkg/input/progress_log.go

```diff
@@ -175,25 +189,48 @@ func (p *progressLog) Write(logBytes []byte) (int, error) {
 			afterFirstLine = true
 		}
 
+		// Safety check: ensure we have at least one element after slice operations
+		if len(p.output) == 0 {
+			// Ensure we have at least 1 line even if p.lines is 0
+			lineCount := p.lines
+			if lineCount == 0 {
+				lineCount = 1
+			}
+			p.output = make([]string, lineCount)
+		}
```

Comment:
Can we DRY this up a bit? Looks like we're duplicating this block a few times...